R/pkg/R/DataFrame.R

```diff
@@ -1266,6 +1266,83 @@ setMethod("dapplyCollect",
             ldf
           })
 
+#' gapply
+#'
+#' Group the SparkDataFrame using the specified columns and apply the R function to each
+#' group.
+#'
+#' @param x A SparkDataFrame
+#' @param cols Grouping columns
+#' @param func A function to be applied to each group partition specified by grouping
+#'             column of the SparkDataFrame. The function `func` takes as argument
+#'             a key - grouping columns and a data frame - a local R data.frame.
+#'             The output of `func` is a local R data.frame.
+#' @param schema The schema of the resulting SparkDataFrame after the function is applied.
```

Comment:
I could have in the documentation smth like: "The schema has to correspond to output SparkDataFrame. It has to be defined for each output column with preferred output column name and corresponding data type."
How does this sound ?